orchard_poseidon_hash.py

```diff
@@ -0,0 +1,48 @@
+from orchard_pallas import Fp
+from orchard_poseidon import perm
+from sapling_utils import leos2ip
+from tv_output import render_args, render_tv
+from tv_rand import Rand
+
+# Initial capacity element
+CAPACITY_ELEMENT = Fp(1 << 65)
+
+def poseidon_hash(x, y):
+    assert isinstance(x, Fp)
+    assert isinstance(y, Fp)
+    return perm([x, y, CAPACITY_ELEMENT])
```

Comment:
The hash function only returns the first element of the state:
```suggestion
    return perm([x, y, CAPACITY_ELEMENT])[0]
```